ui/src/app/applications/components/application-resource-tree/application-resource-tree.tsx

```diff
@@ -189,6 +189,14 @@ function renderResourceNode(props: ApplicationResourceTreeProps, id: string, nod
     }
     const appNode = isAppNode(node);
     const rootNode = !node.root;
+
+    const baseHref = document
```

Comment:
This method is executed very frequently. Accessing DOM here might be pretty expensive. 

Instead of resolving base href here can you please pass it using React context?

The base href already queried on page load here: https://github.com/argoproj/argo-cd/blob/f4400b9493ef730a7d6c992870944a9a7de6a93f/ui/src/app/app.tsx#L20

I would suggest to add baseHref to https://github.com/argoproj/argo-cd/blob/f4400b9493ef730a7d6c992870944a9a7de6a93f/ui/src/app/shared/context.ts#L7 and set it here https://github.com/argoproj/argo-ui/blob/master/src/app/app.tsx#L82

Reply:
Sounds good, thanks for the feedback @alexmt .  As you may be able to tell, I'm certainly no react or frontend dev, so I may need some more help, but I'll try to do what you suggest and try to architect this better!